script.js

```diff
@@ -1,4 +1,811 @@
+  for (let i = 0; i < cardCount; i++) {
+    // Remove card object from the deck and push it to another array of player or pc's cards
+    userCards.push(deck.pop());
+  }
+  return userCards;
+};
+
+// Function to calculate each user's scores, i.e. sum of all their card values
+const calculateScores = function (userCards) {
+  // Initialize sum to 0
+  let sum = 0;
+  // Loop through the user card values and increment the sum by each of the card values
+  for (let i = 0; i < userCards.length; i++) {
+    sum += userCards[i].rank;
+  }
+
+  // Loop through the user cards to check if there is any ace card and if the sum from the previous loop is > 21, if so, reduce sum by 10 for each ace until the sum is <= 21. Examples are available the bottom of this file.
+  // Put it simply, reduce the sum by 10 only if the ace card causes the user to be busted. Ace is counted as 1 instead of 11 in that case
+  for (let j = 0; j < userCards.length; j++) {
+    if (userCards[j].rank === 11 && sum > 21) {
+      sum -= 10;
+    }
+  }
+  return sum;
+};
+
+// Function to show all the cards each user draws in a one line sentence
+const getAllCardsInfo = function (userCards) {
+  // Initialize a string to hold each user's cards info
+  let userCardsInfo = "";
+  for (i = 0; i < userCards.length; i++) {
+    // Update the string by each card name and suit
+    userCardsInfo += `<em>${userCards[i].name}</em> of ${userCards[i].suit},   `;
+  }
+  return userCardsInfo;
+};
+
+// Function to analyze winning conditions and generate result output message
+// First check the strictest condition, i.e. if there is any immediate win by blackjack , then check if anyone has busted, then check the remaining normal conditions
+const getResult = function (
+  playerCards,
+  pcCards,
+  playerScore,
+  pcScore,
+  WINNING_SCORE
+) {
+  // Initialize a string to store result info which will be combined with the standard output message later
+  let result = `<hr> Please drag the slider to bet and press Play to continue playing.`;
+
+  // Video for winning conditions
+  const winningVideo =
+    '<video autoplay loop style="display: block; margin: 0 auto; max-width: 50%; height: 300px;"> <source src="assets/cat.mp4" type="video/mp4" /> </video>';
+
+  // --- Blackjack win ---
+  // If both player and pc have blackjack, i.e. 2 cards on hand with sum of 21
+  if (
+    playerScore === WINNING_SCORE &&
+    playerCards.length === 2 &&
+    pcScore === WINNING_SCORE &&
+    pcCards.length === 2
+  ) {
+    result = `Both you and computer have a blackjack. It's a push!!! ${result}`;
+  } else if (
+    // If only pc has blackjack
+    pcScore === WINNING_SCORE &&
+    pcCards.length === 2
+  ) {
+    result = `Computer has a blackjack. You lost!!! ${result}`;
+  } else if (
+    // If only player has blackjack
+    playerScore === WINNING_SCORE &&
+    playerCards.length === 2
+  ) {
+    result = `You have a blackjack. You win!!! ${winningVideo} ${result}`;
+  }
```

Comment:
This block could possibly be abstracted into its own function `handleBlackjackWin` or something